plugins/vli/fu-vli-pd-device.c

```diff
@@ -590,6 +593,36 @@ fu_vli_pd_device_attach (FuDevice *device, GError **error)
 	return TRUE;
 }
 
+#define VL817_ADDR_GPIO_OUTPUT_ENABLE		0x60A0	/* 0=input, 1=output */
+#define VL817_ADDR_GPIO_SET_OUTPUT_DATA		0x60A1	/* 0=low, 1=high */
+#define VL817_ADDR_GPIO_GET_INPUT_DATA		0x60A2	/* 0=low, 1=high */
+
+static gboolean
+fu_vli_pd_device_attach_vl817_gpiob (FuDevice *device, GError **error)
+{
+	FuVliPdDevice *self = FU_VLI_PD_DEVICE (device);
+	guint8 tmp = 0xff;
+
+	/* set GPIOB output enable */
+	if (!fu_vli_pd_device_read_reg (self, VL817_ADDR_GPIO_OUTPUT_ENABLE,
+					&tmp, error))
+		return FALSE;
+	if (!fu_vli_pd_device_write_reg (self, VL817_ADDR_GPIO_OUTPUT_ENABLE,
+					 tmp | 1 << 1, error))
+		return FALSE;
+
+	/* set GPIOB low to trigger reset */
+	if (!fu_vli_pd_device_read_reg (self, VL817_ADDR_GPIO_SET_OUTPUT_DATA,
+					&tmp, error))
+		return FALSE;
+	if (!fu_vli_pd_device_write_reg (self, VL817_ADDR_GPIO_SET_OUTPUT_DATA,
+					 tmp | 1 << 1, error))
+		return FALSE;
+	fu_device_set_status (device, FWUPD_STATUS_DEVICE_RESTART);
+	fu_device_add_flag (device, FWUPD_DEVICE_FLAG_WAIT_FOR_REPLUG);
+	return TRUE;
+}
```

Comment:
1. GPIOB can only be pulled from a VL817 usbhub, this function should be moved to fu-vli-usbhub-device.c 
2. Enable output bitwise operation L611: tmp |= 0x02
3. Set GPIOB low bitwise operation L619: tmp &= (~0x02)
4. Nitpick L604: guint8 tmp can be init as 0x00 or 0xFF